runtime/Cpp/runtime/src/misc/MurmurHash.h

```diff
@@ -64,12 +65,18 @@ namespace misc {
     template<typename T> // where T is C array type
     static size_t hashCode(const std::vector<Ref<T>> &data, size_t seed) {
       size_t hash = initialize(seed);
-      for (auto entry : data) {
-        hash = update(hash, entry->hashCode());
+      for (auto &entry : data) {
+        hash = update(hash, entry);
```

Comment:
By this change you removed the handling of sub elements hashes. This must stay. Certain objects generated hashes not only from their own state but that of their own data too.

Reply:
This should be delegating to the `update` overload above, which accepts `Ref` and will actually check if its `nullptr`.